linker/shared/src/main/scala/org/scalajs/linker/frontend/modulesplitter/Tagger.scala

```diff
@@ -84,40 +173,47 @@ private class Tagger(infos: ModuleAnalyzer.DependencyInfo,
     }
   }
 
-  private def tag(className: ClassName, pathRoot: ModuleID, pathSteps: List[ClassName]): Unit = {
+  private def tag(className: ClassName, pathRoot: ModuleID, pathSteps: List[ClassName],
+      excludedHopCount: Int, fromExcluded: Boolean): Unit = {
+    val isExcluded = excludedClasses.contains(className)
+
+    val newExcludedHopCount =
+      if (isExcluded && !fromExcluded) excludedHopCount + 1
+      else excludedHopCount
```

Comment:
While I don't think this it invalidates the proof relevance, this is inconsistent with the proof premises, which is
```
∀ (f, c) ∈ E : f ∈ F, c ∈ C ⇒ t(f) < t(c)
```
The way this is computed, what we ensure is instead
```
∀ (c, f) ∈ E : c ∈ C, f ∈ F ⇒ t(c) < t(f)
```
Since changing the proof is probably harder, consider changing the condition instead:
```suggestion
    val newExcludedHopCount =
      if (fromExcluded && !isExcluded) excludedHopCount + 1 // hop from fine to coarse
      else excludedHopCount
```

Reply:
Other than that, LGTM.